File: benchmark/nodejs/lcs.js
const start = Date.now();
const s1 = "abcdefghij".repeat(10);
const s2 = "acegikmoqs".repeat(10);
const m = s1.length, n = s2.length;
const dp = Array.from({ length: m + 1 }, () => Array(n + 1).fill(0));
for (let i = 1; i <= m; i++) {
    for (let j = 1; j <= n; j++) {
        if (s1[i - 1] === s2[j - 1]) dp[i][j] = dp[i - 1][j - 1] + 1;
        else {
            const a = dp[i - 1][j];
            const b = dp[i][j - 1];
            dp[i][j] = a > b ? a : b;
        }
    }
}
const end = Date.now();
console.log(`LCS: ${dp[m][n]}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/python/lcs.py
import time
s1 = "abcdefghij" * 10
s2 = "acegikmoqs" * 10
start = time.time() * 1000
m, n = len(s1), len(s2)
dp = [[0] * (n + 1) for _ in range(m + 1)]
for i in range(1, m + 1):
    for j in range(1, n + 1):
        if s1[i - 1] == s2[j - 1]:
            dp[i][j] = dp[i - 1][j - 1] + 1
        else:
            a = dp[i - 1][j]
            b = dp[i][j - 1]
            dp[i][j] = a if a > b else b
end = time.time() * 1000
print(f"LCS: {dp[m][n]}")
print(f"Time: {end - start:.0f}ms")

